geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/pubsub/AbstractSubscriptionsIntegrationTest.java

```diff
@@ -88,18 +111,91 @@ public void multiSubscribe() {
         .untilAsserted(() -> assertThat(mockSubscriber.getReceivedPMessages()).hasSize(1));
     assertThat(mockSubscriber.getReceivedEvents()).containsExactly("message", "pmessage");
     mockSubscriber.unsubscribe();
-    client.close();
   }
 
   @Test
-  public void unallowedCommandsWhileSubscribed() {
-    Jedis client = new Jedis("localhost", getPort());
+  public void unsubscribingImplicitlyFromAllChannels_doesNotUnsubscribeFromPatterns() {
```

Comment:
It might make sense to move these test to `AbstractPubSubIntegrationTest.java` once [8577](https://issues.apache.org/jira/projects/GEODE/issues/GEODE-8577?filter=allopenissues) is resolved.

Reply:
Do we not already have similar tests in `AbstractPubSubIntegrationTest`? Perhaps `testUnsubscribingImplicitlyFromAllChannels` and `testPunsubscribingImplicitlyFromAllChannels`.

Reply:
Do we not already have similar tests in `AbstractPubSubIntegrationTest`? Perhaps `testUnsubscribingImplicitlyFromAllChannels` and `testPunsubscribingImplicitlyFromAllChannels`

Reply:
We do have a test that only subscribes to channels and then unsubscribes without an argument. We did not have a test that subscribes to channels and psubscribes to patterns, then unsubscribes without an argument (or punsubscribes without an argument).  We were incorrectly unsubscribing from everything (channels and patterns) when punsubscribe or unsubscribe were called without an argument.  Native Redis will only unsubscribe from channels (not patterns) if unsubscribe is called without an argument or punsubscribe only from patterns (not channels) if punsubscribe is called without an argument.